python/ray/tests/test_autoscaler.py

```diff
@@ -2717,24 +2765,40 @@ def testTerminateUnhealthyWorkers(self):
             "min_workers"
         ] = 1
         fill_in_raylet_ids(self.provider, lm)
-        autoscaler.update()
-        # Stopped node metric incremented.
-        mock_metrics.stopped_nodes.inc.assert_called_once_with()
-        # One node left.
-        self.waitForNodes(1)
 
-        # Check the node removal event is generated.
-        autoscaler.update()
-        events = autoscaler.event_summarizer.summary()
-        assert (
-            "Removing 1 nodes of type "
-            "ray-legacy-worker-node-type (lost contact with raylet)." in events
-        ), events
+        if disable_liveness_check:
+            # We've disabled the liveness check, so the unhealthy node should stick
+            # around until someone else takes care of it.
+            # Do several autoscaler updates:
+            for _ in range(10):
+                autoscaler.update()
+            # The nodes are still there.
+            assert self.num_nodes() == 2
```

Comment:
Do we need some sort of synchronization before this is effective?

Reply:
I will add a comment explaining this: Node deletion is synchronous and blocking, as are the methods of the mock node provider used in this test.

The sort of synchronization that would be nice is to make the autoscaler single-threaded in all contexts...

Reply:
Added comments about why no synchronization is required.